core/src/components/datetime/datetime.ios.scss

```diff
@@ -85,27 +85,33 @@
    */
   @include padding($datetime-ios-padding * 0.5, $datetime-ios-padding * 0.5, $datetime-ios-padding * 0.5, $datetime-ios-padding * 0.5);
 
+  align-items: center;
+
   height: calc(100% - #{$datetime-ios-padding});
 }
 
-:host .calendar-day {
-  font-size: 20px;
-}
+:host .calendar-day-wrapper {
+  @include padding(4px);
+
+  // This is required so that the calendar day wrapper
+  // will collapse instead of expanding to fill the button
+  height: 0;
 
-.calendar-day:focus .calendar-day-highlight,
-.calendar-day.calendar-day-active .calendar-day-highlight {
-  opacity: 0.2;
+  min-height: 16px;
 }
 
-.calendar-day.calendar-day-active .calendar-day-highlight {
-  background: current-color(base);
+:host .calendar-day {
+  width: 40px;
+  min-width: 40px;
+
+  height: 40px;
+
+  font-size: 20px;
 }
 
-// !important is needed here to overwrite custom highlight background, which is inline.
-// Does not apply to the active state because highlights aren't applied at all there.
-.calendar-day:focus .calendar-day-highlight {
-  /* stylelint-disable-next-line declaration-no-important */
-  background: current-color(base) !important;
+.calendar-day:focus,
+.calendar-day.calendar-day-active {
+  background: current-color(base, 0.2);
 }
```

Comment:
We will need to rethink this approach because adding `!important` back also overrides any styles set by the user via shadow parts like:

```css
ion-datetime.ios::part(calendar-day):focus {
  background-color: rgb(154 209 98 / 0.2);
}
```

Reply:
Hmm 🤔 I wonder if there's another way to show the `highlightedDates` backgrounds besides inline styling? Maybe it could be set as a CSS variable via `el.style.setProperty()`?

As a fallback, maybe we could tweak the `ios` styles to show a ring around the focused date, similar to `md`?

Reply:
So we could do something like this:

*datetime.tsx*
```diff
style={
  dateStyle && {
-   color: dateStyle.textColor,
-   backgroundColor: dateStyle.backgroundColor,
+   '--day-color': dateStyle.textColor,
+   '--day-background': dateStyle.backgroundColor,
  }
```

*datetime.scss*
```diff
.calendar-day {
-  background: none;
-  color: currentColor;
+  background: var(--day-background);
+  color: var(--day-color);
}
```

This would solve both problems but then it introduces new CSS variables.

Reply:
That would work for me :eyes:

Reply:
Amanda, Liam and I met for a call on this and we decided on the following solution:

1. Change the iOS focus state to use a box shadow to align with MD. This allows us to have a focus state in a way that does not conflict with user styles and does not require the usage of `!important` when setting the focus state internally in datetime.
2. Add `!important` to the inline styles set by `highlightedDates` because the custom CSS shadow parts styling is overriding the `highlightedDates` styles. Both styles are technically coming from a different tree even though the `highlightedDates` styles are set inline inside of the datetime shadow root. This behavior is correct per [WICG/webcomponents#847](https://github.com/WICG/webcomponents/issues/847). A comment has been added in the code explaining this: https://github.com/ionic-team/ionic-framework/blob/e34124a644b030cb8026a9b5e195641dc1878105/core/src/components/datetime/datetime.tsx#L2097-L2108